igor-web/src/main/groovy/com/netflix/spinnaker/igor/travis/TravisBuildMonitor.groovy

```diff
@@ -161,13 +168,14 @@ class TravisBuildMonitor implements PollingMonitor{
 
                             log.info "sending build events for builds between ${lastBuild} and ${currentBuild}"
                             for (int buildNumber = lastBuild+1; buildNumber < currentBuild; buildNumber++) {
-                                Build build = buildMasters.map[master].getBuild(repo, buildNumber) //rewrite to afterNumber list thing
-                                log.info "pushing event for ${master}:${repo.slug}:${build.number}"
-                                String url = "${buildMasters.map[master].baseUrl}/${repo.slug}/builds/${build.id}"
-                                GenericProject project = new GenericProject(repo.slug, new GenericBuild((build.state == BUILD_IN_PROGRESS), build.number, build.duration, TravisResultConverter.getResultFromTravisState(build.state), repo.slug, url))
-
-                                echoService.postEvent(
-                                    new GenericBuildEvent(content: new GenericBuildContent(project: project, master: master, type: 'travis')))
+                                Build build = travisService.getBuild(repo, buildNumber) //rewrite to afterNumber list thing
```

Comment:
@lwander I can add that, but from my tests build did not become null when it was missing. build.number was 0 and the other fields null.